src/qtism/common/datatypes/QtiCoords.php

```diff
@@ -120,52 +120,58 @@ public function inside(QtiPoint $point)
     {
         if ($this->getShape() === QtiShape::DEF) {
             return true;
-        } elseif ($this->getShape() === QtiShape::RECT) {
+        }
+
+        if ($this->getShape() === QtiShape::RECT) {
             return $point->getX() >= $this[0] && $point->getX() <= $this[2] && $point->getY() >= $this[1] && $point->getY() <= $this[3];
-        } elseif ($this->getShape() === QtiShape::CIRCLE) {
+        }
+
+        if ($this->getShape() === QtiShape::CIRCLE) {
             return pow($point->getX() - $this[0], 2) + pow($point->getY() - $this[1], 2) < pow($this[2], 2);
-        } else {
-            // we consider it is a polygon.
-            // - Transform coordinates in vertices.
-            // -- Use of the "point in polygon" algorithm.
-            $vertices = [];
-            for ($i = 0; $i < count($this); $i++) {
-                $vertex = [];
-                $vertex[] = $this[$i]; //x
-                $i++;
-                $vertex[] = $this[$i]; //y
-
-                $vertices[] = $vertex;
+        }
+
+        // we consider it is a polygon.
+        // - Transform coordinates in vertices.
+        // -- Use of the "point in polygon" algorithm.
+        $vertices = [];
+        for ($i = 0, $limit = $this->count(); $i < $limit; $i++) {
```

Comment:
Ok for removing the `count()` from the loop evaluation. But please put it in another line, to keep the `for` readable:

```suggestion
        $limit = $this->count();
        for ($i = 0; $i < $limit; $i++) {
```